crates/biome_css_analyze/src/lint/nursery/no_invalid_position_at_import_rule.rs

```diff
@@ -0,0 +1,99 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{AnyCssRule, CssImportAtRule, CssRuleList};
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow invalid position @import rules.
+    ///
+    /// Any @import rules must precede all other valid at-rules and style rules in a stylesheet (ignoring @charset and @layer), or else the @import rule is invalid.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// a {}
+    /// @import 'foo.css';
+    /// /** ↑
+    /// * This @import */
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// @import 'foo.css';
+    /// a {}
+    /// ```
+    ///
+    pub NoInvalidPositionAtImportRule {
+        version: "next",
+        name: "noInvalidPositionAtImportRule",
+        recommended: true,
+        sources: &[RuleSource::Stylelint("no-invalid-position-at-import-rule")],
+    }
+}
+
+impl Rule for NoInvalidPositionAtImportRule {
+    type Query = Ast<CssRuleList>;
+    type State = CssImportAtRule;
+    type Signals = Option<Self::State>;
```

Comment:
I think this can be a bit better. Instead of highlighting only the first incorrect at-rule, we should highlight ALL incorrect at-rules. For example:

```css
a {}
@import "foo";
@import "bar";
@import "foo";
@import "bar";
```

Here we should flag all import all together.

This means we should return a `Vec<Self::State>. And `State` should be a `TextRange`, we don't need the whole node.

Reply:
Sorry, I didn't know how to do it 🙇‍♂️
`RuleDiagnostic` only accepts a single TextRange.
How should I handle vector format state in `RuleDiagnostic`?

Reply:
The infrastructure does that for you, you don't need to do much more.

You just need to return a vector from the `run` function

Reply:
Thank you for your advice!
I am trying to fix it locally as below.
We add `invalid_import_list` and return it at the end.
But I get a TypeError in the state of RuleDiagnostic::new().


crates/biome_css_analyze/src/lint/nursery/no_invalid_position_at_import_rule.rs
```rs
impl Rule for NoInvalidPositionAtImportRule {
    type Query = Ast<CssRuleList>;
    type State = Vec<TextRange>;   //  ← update
    type Signals = Option<Self::State>;
    type Options = ();

    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
        let node = ctx.query();
        let mut is_invalid_position = false;
        let mut invalid_import_list = Vec::new();  //  ← update

        for rule in node {
            let any_css_at_rule = match rule {
                AnyCssRule::CssAtRule(item) => item.rule().ok(),
                _ => None,
            };

            if let Some(any_css_at_rule) = any_css_at_rule {
                // Ignore @charset, @layer
                if any_css_at_rule.as_css_charset_at_rule().is_some() {
                    continue;
                }
                if any_css_at_rule.as_css_layer_at_rule().is_some() {
                    continue;
                }

                let import_rule = any_css_at_rule.as_css_import_at_rule();
                if let Some(import_rule) = import_rule {
                    if is_invalid_position {
                        invalid_import_list.push(import_rule.range());  //  ← update
                    }
                } else {
                    is_invalid_position = true;
                }
            } else {
                is_invalid_position = true;
            }
        }

        if invalid_import_list.len() > 0 {
            return Some(invalid_import_list);  //  ← update
        }

        None
    }

    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
        Some(
            RuleDiagnostic::new(
                rule_category!(),
                state,  //  ← TypeError 
                markup! {
                    "This "<Emphasis>"@import"</Emphasis>" is in the wrong position."
                },
            )
            .note(markup! {
                "Any "<Emphasis>"@import"</Emphasis>" rules must precede all other valid at-rules and style rules in a stylesheet (ignoring @charset and @layer), or else the "<Emphasis>"@import"</Emphasis>" rule is invalid."
            }).note(markup! {
                "Consider moving import position."
            })
        )
    }
}
```

Reply:
Signal must return a `Vec<Self::State>` instead of `Option<Self::State>`. 

Basically, you must always return a vector.

Reply:
@ematipico 

Thank you for your advice!
https://github.com/biomejs/biome/pull/2717/commits/6f752eded56e1130c35d5f49d23d629e7bc3b8be

Please also check this 🙇‍♂️ 
https://github.com/biomejs/biome/pull/2717#discussion_r1590284228